### PR TITLE
feat: Add validation of entities to `breezy-hr` maps

### PR DESCRIPTION
**Pull request by martinalbert**
## Description
Adds operations for checking availability of entities needed to perform the use case.

## Motivation and Context
Breezy-hr returns in most endpoints only 400, 401 and other errors as 500.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by martinalbert** [comment]
@janhalama workable also doesn't offer much detail into what is missing, should I open similar PR for validation of entities in workable?

File: grid/recruitment/create-lead/maps/breezy-hr.suma [hunk 1]
```diff
@@ -274,7 +282,7 @@ operation CheckJobExists {
       // Breezy HR returns http status code 500 in case that the job ID doesn't exist. It could be caused by other reasons as well, but at least the error helps the client understand the error better.
       fail {
         title = 'Not Found'
-        detail = 'Unable to get job ID ' + args.jobId + '. Check that the job ID exists.'
+        detail = 'Unable to get job under ID ' + args.jobId + '. Check that the job with that ID exists.'
```
Comment (janhalama) [inline grid/recruitment/create-lead/maps/breezy-hr.suma:285]:
```suggestion
        detail = 'Unable to get job by ID ' + args.jobId + '. Check that the job with that ID exists.'
```

File: grid/recruitment/create-lead/maps/breezy-hr.suma [hunk 2]
```diff
@@ -299,3 +307,45 @@ operation CheckJobExists {
...
+    }
+
+    response 200 {       
+      return {}
+    }
+
+    response 500 "application/json" {
+      fail {
+        title = 'Not Found'
+        detail = 'Unable to get company under ID ' + args.companyId + '. Check that the company with that ID exists.'
```
Comment (janhalama) [inline grid/recruitment/create-lead/maps/breezy-hr.suma:327]:
```suggestion
        detail = 'Unable to get company by ID ' + args.companyId + '. Check that the company with that ID exists.'
```

**Comment by janhalama** [comment]
> @janhalama workable also doesn't offer much detail into what is missing, should I open similar PR for validation of entities in workable?

Yes that would be great.

File: grid/recruitment/create-candidate/maps/breezy-hr.suma [hunk 3]
```diff
@@ -2,11 +2,26 @@ profile = "recruitment/create-candidate@1.0"
 provider = "breezy-hr"
 
 map CreateCandidate {
+  call CheckCompanyExists (
```
Comment (janhalama) [inline grid/recruitment/create-candidate/maps/breezy-hr.suma:5]: I miss test case for not existing company ID.

**Review by janhalama — COMMENTED** [review]
I miss tests for not existing company ID and I propose little error description fixes.

**Comment by martinalbert** [comment]
@janhalama added missing tests and opened new PR built on top of this one for validation of entities in `workable` maps.

**Review by janhalama — APPROVED** [review]
LGTM

**Pull request merged**